### PR TITLE
adds moped_user_saved_views_to_database

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/19686

## Testing
**URL to test:** 

local

**Steps to test:**
- restart cluster or apply migrations and metadata
- check to see that the `moped_user_saved_views` table exists and that the columns and permissions look correct
- check for the `set_moped_user_saved_views_updated_at` trigger function and the activity log transformation

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- ~[ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

### CODE REVIEW

**Review by chiaberry — APPROVED** [review]
I see the table, tested down migration and 👍🏼

**Review by johnclary — APPROVED** [review]
looks great! i had a few questions 🚢 🚢 🚢

File: moped-database/metadata/tables.yaml
Comment (mddilley) [inline moped-database/metadata/tables.yaml:5284]: @tillyw could you remove this event trigger? I don't imagine us showing saved view activity anywhere (yet!) since our activity log is on the project level. Thanks!

Let me know if you had something else in mind or if I'm missing an important part. 🙏

**Review by mddilley — CHANGES_REQUESTED** [review]
Looking great! I was able to test an insert and see everything working besides a couple small things. Thanks so much, and looking forward to the rest of this coming together! 🔖

File: moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql
Comment (mddilley) [reply moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql:21]: forgot to note that i think this is a great idea. One of the nice code details that came out of the VZ revamp work. Tilly - let me know if I can help with this.

File: moped-database/metadata/tables.yaml
Comment (tillyw) [reply moped-database/metadata/tables.yaml:5284]: no that makes sense! thank you for pointing that out :)

File: moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql
Comment (tillyw) [reply moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql:25]: agreed, these comments could be helpful to refer to in the future!

**Comment by tillyw** [comment]
thanks, y'all! i've made the requested changes and will merge once i get re-approval from @mddilley :)

File: moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql
Comment (mddilley) [inline moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql:5]: @tillyw Could you update this `query_filters` column to be `NOT NULL` too? Just saw this when browsing the DB

File: moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql
Comment (mddilley) [inline moped-database/migrations/1730324794396_add_moped_user_saved_views_table/up.sql:25]: Small typo in the column name here - could you fix? Love these comments by the way! 🙏 ❤️

**Review by mddilley — COMMENTED** [review]
Two tiny changes, and then I'll approve asap. Thank you! 🙏

**Comment by tillyw** [comment]
fixed! i'll merge once i get re-approved :)

**Review by mddilley — APPROVED** [review]
I retested this for good measure, and I see inserts and updates working just as expected. Thanks for adding those last details!

**Comment by mddilley** [comment]
dropping these SQL statements that i used to test since i lost mine from last time around. 🔖

```sql
INSERT INTO moped_user_saved_views (is_deleted, query_filters, description, url, created_by_user_id, updated_by_user_id)
		VALUES(FALSE, '[]', 'Lamar projects', 'https://localhost:3000/moped/projects?filters=%5B%7B%22field%22%3A%22project_name_full%22%2C%22operator%22%3A%22string_contains_case_insensitive%22%2C%22value%22%3A%22lamar%22%7D%5D&isOr=false', 1, 1);

UPDATE moped_user_saved_views SET description = 'updated' WHERE id = 2;
```

**Pull request merged**
